gui/odmr/odmrgui.py

```diff
@@ -366,6 +476,179 @@ def _menu_settings(self):
+
+        # stop
+        stop_label = QtWidgets.QLabel(groupBox)
+        stop_label.setText('Stop:')
+        setattr(self._mw.odmr_control_DockWidget, 'stop_label_{}'.format(insertion_row), stop_label)
+        stop_freq_DoubleSpinBox = ScienDSpinBox(groupBox)
+        stop_freq_DoubleSpinBox.setMaximum(constraints.max_frequency)
+        stop_freq_DoubleSpinBox.setMinimum(constraints.min_frequency)
+        stop_freq_DoubleSpinBox.setMinimumSize(QtCore.QSize(80, 0))
+        stop_freq_DoubleSpinBox.setValue(self._odmr_logic.mw_stops[0])
+        stop_freq_DoubleSpinBox.setMinimumWidth(75)
+        stop_freq_DoubleSpinBox.setMaximumWidth(100)
+        stop_freq_DoubleSpinBox.editingFinished.connect(self.change_sweep_params)
+        setattr(self._mw.odmr_control_DockWidget, 'stop_freq_DoubleSpinBox_{}'.format(insertion_row),
+                stop_freq_DoubleSpinBox)
+
+        gridLayout.addWidget(stop_label, insertion_row, 5, 1, 1)
+        gridLayout.addWidget(stop_freq_DoubleSpinBox, insertion_row, 6, 1, 1)
+
+        starts = self.get_frequencies_from_spinboxes('start')
+        stops = self.get_frequencies_from_spinboxes('stop')
+        steps = self.get_frequencies_from_spinboxes('step')
+        power = self._mw.sweep_power_DoubleSpinBox.value()
+
+        self.sigMwSweepParamsChanged.emit(starts, stops, steps, power)
+        self._mw.fit_range_SpinBox.setMaximum(self._odmr_logic.ranges)
+        self._mw.odmr_control_DockWidget.matrix_range_SpinBox.setMaximum(self._odmr_logic.ranges)
+        self._odmr_logic.ranges += 1
+
+        # remove stuff that remained from the old range that might have been in place there
+        key = 'channel: {0}, range: {1}'.format(self.display_channel, self._odmr_logic.ranges - 1)
+        if key in self._odmr_logic.fits_performed:
+            self._odmr_logic.fits_performed.pop(key)
+        return
+
+    def remove_ranges_gui_elements_clicked(self):
+        if self._odmr_logic.ranges == 1:
+            return
+
+        remove_row = self._odmr_logic.ranges - 1
+
+        groupBox = self._mw.odmr_control_DockWidget.ranges_groupBox
+        gridLayout = groupBox.layout()
+
+        object_dict = self.get_objects_from_groupbox_row(remove_row)
+
+        for object_name in object_dict:
+            if 'DoubleSpinBox' in object_name:
+                object_dict[object_name].editingFinished.disconnect()
+            object_dict[object_name].hide()
+            gridLayout.removeWidget(object_dict[object_name])
+            del self._mw.odmr_control_DockWidget.__dict__[object_name]
+
+        starts = self.get_frequencies_from_spinboxes('start')
+        stops = self.get_frequencies_from_spinboxes('stop')
+        steps = self.get_frequencies_from_spinboxes('step')
+        power = self._mw.sweep_power_DoubleSpinBox.value()
+        self.sigMwSweepParamsChanged.emit(starts, stops, steps, power)
+
+        self._odmr_logic.range_to_fit -= 1
```

Comment:
If range_to_fit is zero this might set a negative value and will eventually cause an error when fitting. Simply reset to 0 each time or check whether is it already 0.